### PR TITLE
[NAV] Restrict MC RoC/RoD rate-limiting to the case where we actually accelerate

### PR DESCRIPTION
**Pull request by digitalentity**
Fixes https://github.com/iNavFlight/inav/issues/4240

### CODE REVIEW

File: src/main/navigation/navigation_multicopter.c [hunk 1]
```diff
@@ -74,9 +74,10 @@ static void updateAltitudeVelocityController_MC(timeDelta_t deltaMicros)
 
     posControl.pids.pos[Z].output_constrained = targetVel;
 
-    // limit max vertical acceleration to 1/5G (~200 cm/s/s) if we are increasing velocity.
+    // limit max vertical acceleration to 1/5G (~200 cm/s/s) if we are increasing RoC or RoD (only if vel is of the same sign)
     // if we are decelerating - don't limit (allow better recovery from falling)
-    if (fabsf(targetVel) > fabsf(posControl.desiredState.vel.z)) {
+    const bool isSameDirection = (targetVel > 0 && posControl.desiredState.vel.z > 0) || (targetVel < 0 && posControl.desiredState.vel.z < 0);
```
Comment (fiam) [inline src/main/navigation/navigation_multicopter.c]: What do you think about using signbit here?

Comment (digitalentity) [reply src/main/navigation/navigation_multicopter.c]: signbit will be less practical since it returns zero for both "0" and positive numbers.

Comment (ghost) [reply src/main/navigation/navigation_multicopter.c]: @digitalentity If your confident that the patch will do the job.
I'm willing to give it a test, before 2.1 is released. Just to be sure.

Target - Omnibus F4 Pro

**Comment by digitalentity** [comment]
@kardon18 here you are 
[inav_2.1.0_OMNIBUSF4PRO.hex.zip](https://github.com/iNavFlight/inav/files/2788705/inav_2.1.0_OMNIBUSF4PRO.hex.zip)

**Comment by kardon18** [comment]
Thanks Konstantin :)
As can be seen from the log. It only drops the throttle as low as 35% now. And only for 1 frame of the DVR footage. So it most certainly doesn't loose any altitude now.
Although. (without meaning to sound like I'm ungrateful). It does spike in throttle as can be seen.
I only mention this. In case you can tweak it a little more, to make it perfect :)  But if not...Then that's fine. I am happy it works better than before.

Thank you once again


changes

**Comment by ghost** [comment]
Thanks Konstantin :)
As can be seen from the log. It only drops the throttle as low as 35% now. And only for 1 frame of the DVR footage. So it most certainly doesn't loose any altitude now.
It does peak a little before it levels off to climb in EXTRA altitude mode. But I am happy it works better than before.

Thank you once again


changes

**Comment by digitalentity** [comment]
This fix doesn't actually "fix" the issue - that would be impossible without rethinking the "takeoff" procedure. This fix merely minimizes the effect of the bug. Not perfect, but much safer than before.

Throttle increase is expected - target RTH altitude is higher than current altitude and quad needs to build up climb rate to reach it. It also needs to compensate for that drop in throttle.

**Comment by digitalentity** [comment]
Thanks for testing. I'm going to merge this now.

**Pull request merged**